### ISSUE TITLE
Test failures in latest `woofmark` update

### ISSUE THREAD

**Issue by jywarren** [publiclab/PublicLab.Editor#607]
In https://github.com/publiclab/PublicLab.Editor/pull/597 we started seeing test failures - @keshav234156 opened https://github.com/publiclab/PublicLab.Editor/pull/602 to revert. 

However, the test failures don't trigger a Travis failure, so we weren't seeing them; see this for example: 

https://travis-ci.com/github/publiclab/PublicLab.Editor/jobs/379516209

```
     ReferenceError: Can't find variable: PL in file:///home/travis/build/publiclab/PublicLab.Editor/spec/javascripts/button_spec.js (line 7) (1)
     Expected false to be true. (2)
```

It looks like potentially a syntax error in our Jasmine tests, under `jasmine:publiclabeditor`. 

The commits were:

Bumps [woofmark](https://github.com/jywarren/woofmark) from `12e2c1d` to `39237b7`.

<ul>
<li><a href="https://github.com/jywarren/woofmark/commit/39237b798308e6855acde1a938c18a8e4da62fce"><code>39237b7</code></a> heading test (<a href="https://github-redirect.dependabot.com/jywarren/woofmark/issues/34">#34</a>)</li>
<li><a href="https://github.com/jywarren/woofmark/commit/7a47ab75957588a254f040bc244e1d29f05657db"><code>7a47ab7</code></a> Fix .gitpod.yml syntax (missing '-' in tasks) (<a href="https://github-redirect.dependabot.com/jywarren/woofmark/issues/28">#28</a>)</li>
<li><a href="https://github.com/jywarren/woofmark/commit/3a346a6feac536560c55a353eb8a12f88fc643a4"><code>3a346a6</code></a> Bump jest from 26.1.0 to 26.2.2 (<a href="https://github-redirect.dependabot.com/jywarren/woofmark/issues/36">#36</a>)</li>
<li><a href="https://github.com/jywarren/woofmark/commit/a5b9afa6d47b93b0d6052553f737c638c1ed0689"><code>a5b9afa</code></a> Bump browserify from 8.1.0 to 16.5.2 (<a href="https://github-redirect.dependabot.com/jywarren/woofmark/issues/38">#38</a>)</li>
<li><a href="https://github.com/jywarren/woofmark/commit/a35205ae1d3c7b80546aa18427af2657d7cc0fa3"><code>a35205a</code></a> Bump jshint from 2.11.1 to 2.12.0 (<a href="https://github-redirect.dependabot.com/jywarren/woofmark/issues/37">#37</a>)</li>
<li><a href="https://github.com/jywarren/woofmark/commit/8160220205e6ba082b1fa11a28c00ceed3075bd8"><code>8160220</code></a> Replacing uglify-js with Terser (<a href="https://github-redirect.dependabot.com/jywarren/woofmark/issues/29">#29</a>)</li>
<li><a href="https://github.com/jywarren/woofmark/commit/e6f56ea701b4c74a0e0c1096f056ee8f4fe4186f"><code>e6f56ea</code></a> fix bold texts loosing formatting after converting to markdown and back to wy...</li>
<li><a href="https://github.com/jywarren/woofmark/commit/e9e7bb466e42c616662a118e8b461e467cc25b78"><code>e9e7bb4</code></a> add dependencies for tests and modify jest.config (<a href="https://github-redirect.dependabot.com/jywarren/woofmark/issues/17">#17</a>)</li>
<li><a href="https://github.com/jywarren/woofmark/commit/9bd0750a264ef9ff0ce4d43ed5881e7f9ed52b58"><code>9bd0750</code></a> Install gitpod for auto-built previews on each PR (<a href="https://github-redirect.dependabot.com/jywarren/woofmark/issues/16">#16</a>)</li>
<li><a href="https://github.com/jywarren/woofmark/commit/4f8d7654ba10c6636b2180299ed5e656e61cfffe"><code>4f8d765</code></a> Merge pull request <a href="https://github-redirect.dependabot.com/jywarren/woofmark/issues/13">#13</a> from NitinBhasneria/latestChanges</li>
<li>Additional commits viewable in <a href="https://github.com/jywarren/woofmark/compare/12e2c1d4559268e2de605ad6084d5dce24ea42c6...39237b798308e6855acde1a938c18a8e4da62fce">compare view</a></li>
</ul>

Let's take a close look and see what could have caused this. Notice that Jest tests did not fail.

**Comment by jywarren** [comment]
Is this due to Jasmine being unable to read es6, perhaps? @sagarpreet-chadha @Shulammite-Aso @shreyaa-sharmaa @NitinBhasneria @keshav234156 - any thoughts? I notice that we added Terser to woofmark in the meantime?

**Comment by keshav234156** [comment]
Actually it's difficult to find as we are to find only when we integrate it with Editor. It's due to update that we have made in woofmark

**Comment by jywarren** [comment]
I'm not sure ... the fact that Jasmine is not finding it at all seems to be a big hint. And yet it passes Jest tests. 

We haven't really changed the Jasmine setup, i think? https://github.com/publiclab/PublicLab.Editor/tree/main/spec/javascripts (one change 18 days ago)

This to me suggests that it's a difference in the environment somehow. I know Jasmine has trouble with ES6 and we don't have Babel installed to deal with this... ?

Also noting connection to a previous issue with Travis here which ended up being related to Node version: https://github.com/publiclab/PublicLab.Editor/issues/439 - and there, we asked `@HarshKhandeparkar` and `@VladimirMikulic` for help. If we get really stuck let's see if they have suggestions, as they are really good at this stuff?

Thanks!

**Comment by sagarpreet-chadha** [comment]
I agree with @jywarren , it maybe due to `es6` usage in upstream library - `woofmark`.
I saw that we are now telling `jshint` that the code is written in `es6` here:
https://github.com/jywarren/woofmark/blob/39237b798308e6855acde1a938c18a8e4da62fce/.jshintrc#L2

Maybe this is causing issue?

**Comment by sagarpreet-chadha** [comment]
I think it is due to use of `const` keyword which is es6 construct (I only asked asked people to change `var` to `const` in some PR's)

**Comment by sagarpreet-chadha** [comment]
We are using `const` and even `let` also here https://github.com/jywarren/woofmark/commit/e6f56ea701b4c74a0e0c1096f056ee8f4fe4186f

@Shulammite-Aso would you like to test if this is the issue?

There are 2 things we can do now in `woofmark`:
1. Remove `let` and `const` everywhere and then try installing this branch in PL.editor by changing `package.json` (linking woofmark to your branch) and then running Jasmine tests locally.

2. Setup a transpiler like `babel` which will first convert `ES6` to lower ECMA version and then later bundle them into `/dist` folder. (This we should do anyways)

Thanks :smile:

**Comment by jywarren** [comment]
Awesome. I guess I feel like option 2 makes sense, but it is a bit of work. @sagarpreet-chadha has actually done this (although i didn't review it 😭 ) in this PR so you can see how it's usually configured!

https://github.com/publiclab/leaflet-environmental-layers/pull/240/files

**Comment by Shulammite-Aso** [comment]
Hi @sagarpreet-chadha @jywarren is anybody working on this yet?
Things must have gotten over my head. I thought i read you were working on this @sagarpreet-chadha ?
Just reading through again and saw that wasn't what you said.

I should try setting this up if nobody is already on it, and hit you up @sagarpreet-chadha if i have any question.

**Comment by jywarren** [comment]
Thank you so much! we had gotten a little behind on this but your help is SO APPRECIATED. This is a big blocker! Looking at your solution now. Thanks!! 🎉
